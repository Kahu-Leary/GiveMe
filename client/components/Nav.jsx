import { logOff } from 'authenticare/client/auth'
import React from 'react'
import { Link } from 'react-router-dom'
import { IfAuthenticated, IfNotAuthenticated } from './Authenticated'
import SearchBar from './SearchBar'

import CategoriesMenu from './CategoriesMenu'

function Nav() {
  return (
    <div className='alingElementNav'>
      <img className="nav-logo" src='./img/logo-txt-white.png' />

      <div className='alingSearchBar'>
        <SearchBar />
        <CategoriesMenu />
      </div>

      <div className="navLinks">
        <IfAuthenticated>
          <Link className="link" to="/ListAnItem">Add Item</Link>
        </IfAuthenticated>
        <IfNotAuthenticated>
          <Link className="link" to="/signin">Add Item</Link>
        </IfNotAuthenticated>

        <Link className="link" to="/Home">Home</Link>
        <Link className="link" to="/About">About</Link>
        <Link className="link" to="/ContactUs">Contact Us</Link>
        <IfAuthenticated>
          <Link className="link" to="/signin" onClick={logOff}>Log out</Link>
        </IfAuthenticated>
        <IfNotAuthenticated>
          <Link className="link" to="/SignIn">Sign In</Link>
        </IfNotAuthenticated>
      </div>
    </div >
  )
}
export default Nav
