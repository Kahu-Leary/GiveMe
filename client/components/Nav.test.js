import React from 'react'
// import { HashRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

import Nav from './Nav'

it('renders correctly', () => {
  const tree = renderer
    .create(
    //  ? <HashRouter>
      <Nav/>
    //   </HashRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
