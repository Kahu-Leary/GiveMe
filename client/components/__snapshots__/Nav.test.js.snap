// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="alingElementNav"
>
  <img
    className="nav-logo"
    src="./img/logo-txt-orange.png"
  />
  <div
    className="alingSearchBar"
  >
    <input
      className="input is-rounded is-small searchBar "
      id="searchbar "
      name="searchbar"
      placeholder="Search"
      type="email"
    />
    <div>
      <div
        className="dropdown"
      >
        <div
          className="dropdown-trigger"
        >
          <button
            aria-controls="dropdown-menu3"
            aria-haspopup="true"
            className="button primaryBtn "
            onClick={[Function]}
          >
            <span>
              Categories
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu"
          id="dropdown-menu3"
          role="menu"
        >
          <div
            className="dropdown-content"
          >
            <a
              className="dropdown-item"
              href="#"
            >
              Clothing
            </a>
            <a
              className="dropdown-item"
              href="#"
            >
              Books
            </a>
            <a
              className="dropdown-item"
              href="#"
            >
              Outdoor
            </a>
            <a
              className="dropdown-item"
              href="#"
            >
              Eletronics
            </a>
            <a
              className="dropdown-item"
              href="#"
            >
              Homewear
            </a>
            <a
              className="dropdown-item"
              href="#"
            >
              Arts
            </a>
            <a
              className="dropdown-item"
              href="#"
            >
              Eletronics
            </a>
            <a
              className="dropdown-item"
              href="#"
            >
              Gamin
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
